### PR TITLE
Search by year bug

### PR DESCRIPTION
**Pull request by jpc20**
Adds a new route `/advanced/year` to search by only the year

### CODE REVIEW

File: app/services/nytimes_service.rb [hunk 1]
```diff
@@ -47,19 +46,24 @@ def get_most_recent_obituaries
...
-
     response = conn.get do |req|
       req.params['fq'] = "section_name:(obituaries) AND headline:(#{name}) AND pub_year:(#{year})"
     end
     JSON.parse(response.body, symbolize_names: true)
   end
 
+  def get_obituaries_by_year(year)
+    response = conn.get do |req|
+      req.params['fq'] = "section_name:(obituaries) AND pub_year:(#{year})"
```
Comment (jpc20) [inline app/services/nytimes_service.rb:57]: This is the only new method I made. It only requests the year with no name

File: app/lib/obituary_microservice.rb [hunk 2]
```diff
@@ -31,4 +31,9 @@ class ObituaryMicroservice < Sinatra::Base
     obituaries = NytimesResults.new.get_advanced_obituaries(params[:name], params[:date])
     ObituarySerializer.new.format_obituaries(obituaries).to_json
   end
+
+  get '/advanced/year' do
+    obituaries = NytimesResults.new.get_obituaries_by_year(params[:date])
+    ObituarySerializer.new.format_obituaries(obituaries).to_json
+  end
```
Comment (stellakunzang) [inline app/lib/obituary_microservice.rb:38]: thank you! I didn't realize that I couldn't just send whatever I want willy nilly. This experience has better prepared me to write the microservice README with all the endpoints.

**Review by stellakunzang — APPROVED** [review]
Boom Boom approved!

**Pull request merged**